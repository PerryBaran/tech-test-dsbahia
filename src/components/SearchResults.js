import React from "react";
import "../styles/searchResults.css";

const SearchResults = ({ results }) => {
  return (
    <div className="search-results-container">
      <div className="image-grid">
        {results.map((image) => (
          <img
            key={image}
            src={image}
            alt="spaceImage"
            className="search-result-image"
          />
        ))}
      </div>
    </div>
  );
};

export default SearchResults;
