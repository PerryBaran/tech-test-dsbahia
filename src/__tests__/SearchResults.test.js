import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const mockResults = [
    "https://images-assets.nasa.gov/image/PIA12235/PIA12235~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA12228/PIA12228~thumb.jpg",
  ];

  it("Renders Correctly", () => {
    const { asFragment } = render(<SearchResults results={mockResults} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
