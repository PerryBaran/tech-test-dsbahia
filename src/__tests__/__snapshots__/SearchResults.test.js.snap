// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Renders Correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results-container"
  >
    <div
      class="image-grid"
    >
      <img
        alt="spaceImage"
        class="search-result-image"
        src="https://images-assets.nasa.gov/image/PIA12235/PIA12235~thumb.jpg"
      />
      <img
        alt="spaceImage"
        class="search-result-image"
        src="https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg"
      />
      <img
        alt="spaceImage"
        class="search-result-image"
        src="https://images-assets.nasa.gov/image/PIA12228/PIA12228~thumb.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
